### PR TITLE
feat(expr): support to_timestamp(text,text)

### PR DESCRIPTION
**Pull request by TennyZhuang**
Signed-off-by: TennyZhuang <zty0826@gmail.com>

I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

support to_timestamp(text,text)

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* SQL commands, functions, and operators

### Release note

* support to_timestamp(text,text)

## Refer to a related PR or issue link (optional)

#6894

### CODE REVIEW

**Comment by lmatz** [comment]
There are some test cases at https://github.com/risingwavelabs/risingwave/blob/main/src/tests/regress/data/sql/horology.sql#L354 that can be used to validate if they are helpful

**Comment by TennyZhuang** [comment]
> There are some test cases at https://github.com/risingwavelabs/risingwave/blob/main/src/tests/regress/data/sql/horology.sql#L354 that can be used to validate if they are helpful

I'm sure only few of them can be supported, we can make them complete later.

**Comment by TennyZhuang** [comment]
There are too many differences between pg and chrono :)

**Review by wangrunji0408 — COMMENTED** [review]
LGTM for other parts except `to_timestamp_const_tmpl` function. 🤯

File: proto/expr.proto [hunk 1]
```diff
@@ -43,9 +43,14 @@ message ExprNode {
...
     EXTRACT = 101;
     TUMBLE_START = 103;
+    // From f64 to timestamp.
+    // e.g. `select to_timestamp(1672044740.0)`
     TO_TIMESTAMP = 104;
     AT_TIME_ZONE = 105;
     DATE_TRUNC = 106;
+    // Parse text to timestamp by format string.
+    // e.g. `select to_timestamp('2022 08 21', 'YYYY MM DD')`
+    TO_TIMESTAMP1 = 107;
```
[the remark this answers is not in the document]
Comment (TennyZhuang) [reply proto/expr.proto:53]: Whatever, oh has so many overloads, not only by the number of rags.

File: src/expr/src/vector_op/to_timestamp.rs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    chrono::format::parse(&mut parsed, s, tmpl.borrow_items().iter())?;
+    if parsed.year.is_none()
+        && parsed.year_div_100.is_none()
+        && parsed.year_mod_100.is_none()
+        && parsed.isoyear.is_none()
+        && parsed.isoyear_div_100.is_none()
+        && parsed.isoyear_mod_100.is_none()
+    {
+        parsed.set_year(-1).unwrap();
+    }
```
Comment (BowenXiao1999) [inline src/expr/src/vector_op/to_timestamp.rs:33]: Add some comments for this part?

**Review by BowenXiao1999 — APPROVED** [review]
Rest LGTM

**Pull request merged**